ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -422,14 +466,16 @@ private TransactionSelectionResult handleTransactionNotSelected(
     final var pendingTransaction = evaluationContext.getPendingTransaction();
 
     // check if this tx took too much to evaluate, and in case it was invalid remove it from the
-    // pool, otherwise penalize it.
+    // pool, otherwise penalize it. Not synchronized since there is no state change here.
     final TransactionSelectionResult actualResult =
         isTimeout.get()
             ? transactionTookTooLong(evaluationContext, selectionResult)
                 ? selectionResult.discard()
                     ? INVALID_TX_EVALUATION_TOO_LONG
                     : TX_EVALUATION_TOO_LONG
-                : BLOCK_SELECTION_TIMEOUT
+                : selectionResult.discard()
+                    ? BLOCK_SELECTION_TIMEOUT_INVALID_TX
```

Comment:
Isn't this supposed to be a dead branch? On line 473 we evaluated that `selectionResult.discard()` is false and only in this case we can go to evaluation on line 476 or am I misinterpreting this?

Reply:
I figured it out. I misinterpreted the indentation